engine/image/router.go

```diff
@@ -222,3 +223,89 @@ func (s *imageRouter) getImagesGet(ctx context.Context, w http.ResponseWriter, r
 
 	return nil
 }
+
+
+func (s *imageRouter) getImageHistory(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
+	imageName := vars["name"]
+	filePath, ok := s.images[imageName]
+	if !ok {
+		return errdefs.NotFound(xerrors.Errorf("unknown image: %s", imageName))
+	}
+
+	opener := func() (io.ReadCloser, error) {
+		return tarfile.Open(filePath)
+	}
+
+	img, err := tarball.Image(opener, nil)
+	if err != nil {
+		return errdefs.NotFound(xerrors.Errorf("unable to open the file path (%s): %w", filePath, err))
+	}
+
+	config, err := img.ConfigFile()
+	if err != nil {
+		return errdefs.Unavailable(err)
+	}
+
+	rc, err := tarfile.Open(filePath)
+	defer rc.Close()
+	if err != nil {
+		return errdefs.NotFound(xerrors.Errorf("unable to open the file path (%s): %w", filePath, err))
+	}
+
+	b, err := tarfile.ExtractFileFromTar(rc, "manifest.json")
+	if err != nil {
+		return errdefs.Unavailable(err)
+	}
+
+	var manifests tarball.Manifest
+	if err := json.Unmarshal(b, &manifests); err != nil {
+		return errdefs.Unavailable(err)
+	}
+
+	if len(manifests) != 1 {
+		return errdefs.Unavailable(xerrors.New("tarball must contain only a single image to be used with testdocker"))
+	}
+	var allLayerSizes []int64
+	for _, manifest := range manifests{
+		fmt.Println(manifest.Layers)
+		for _, layerPath := range manifest.Layers{
```

Comment:
Why don't you use Layer()?
https://pkg.go.dev/github.com/google/go-containerregistry@v0.6.0/pkg/v1#Image